### PR TITLE
CosmosDBMemory Store Serialization Fix, Documentation, and Improved Factory Constructor

### PR DESCRIPTION
**Pull request by awharrison-28**
### Motivation and Context
This PR to the feature branch was missed before merging to main. These are critical fixes to the CosmosDBMemoryStore connector. 

### Description
- removed `CosmosDBMemoryRecord` dependency on kernel `Embeddings` struct 
- factory constructor must be static
- `DoesCollectionExistAsync()` must always return false to maintain consistency with existing `SemanticTextMemory` operation. CosmosDB does not support checking if a container exists without creating it. MemoryCollection creation optimizations will be addressed in future PRs.
- Serialization and Deserialization of `CosmosDBMemoryRecords` simplified using System.Text.Json and built-in CosmosDB `ReadItemAsync<>`
-Identified where necessary that partitionKey is not used for lookups

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by lemillermicrosoft — DISMISSED** [review]
LGTM

File: dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryRecord.cs [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
 using System;
-using System.Collections.Generic;
-using Microsoft.SemanticKernel.Memory;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Serialization;
 
 namespace Microsoft.SemanticKernel.Connectors.Memory.Cosmos;
 
 /// <summary>
-/// A Cosmos memory record.
```
Comment (adrianwyatt) [inline dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryRecord.cs:12]: I switched from "CosmosDB" to "Cosmos" to follow the Azure naming conventions for Cosmos DB.

File: dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs [hunk 2]
```diff
@@ -95,64 +99,53 @@ public async Task CreateCollectionAsync(string collectionName, CancellationToken
     /// <inheritdoc />
     public Task<bool> DoesCollectionExistAsync(string collectionName, CancellationToken cancel = default)
     {
-        var container = this._database.Client.GetContainer(this._databaseName, collectionName);
-        return Task.FromResult(container != null);
+        // Azure Cosmos DB does not support checking if container exists without attempting to create it. This does not break the interface but it is not ideal.
```
Comment (adrianwyatt) [inline dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs:102]: Is creating containers idempotent? If so, that would be good to mention here.

File: dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs [hunk 3]
```diff
@@ -95,64 +99,53 @@ public async Task CreateCollectionAsync(string collectionName, CancellationToken
...
-
-        if (response.StatusCode == HttpStatusCode.OK)
-        {
-            this._log.LogInformation("Collection {0} deleted", collectionName);
-        }
-        else
-        {
-            throw new CosmosException("Unable to delete collection", response.StatusCode, 0, collectionName, 0);
-        }
+        _ = await container.DeleteContainerAsync(cancellationToken: cancel);
```
Comment (adrianwyatt) [inline dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs:110]: Does this throw on unsuccessful status codes?

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryRecord.cs (hunk 1 above)
Comment (awharrison-28) [reply dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryRecord.cs]: renamed CosmosDB -> Cosmos

File: dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs (hunk 2 above)
Comment (adrianwyatt) [reply dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs]: Could you do a container get? https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.container.deletecontainerasync?view=azure-dotnet

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs (hunk 3 above)
Comment (awharrison-28) [reply dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs]: added try-catch, notable exception is Container Not Found

File: dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs (hunk 2 above)
Comment (awharrison-28) [reply dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs]: unfortunately no :( the behavior is mentioned in the comment

**Review by adrianwyatt — APPROVED** [review]

**Pull request merged**